### PR TITLE
filter for custom and system pods

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2334.

### CODE REVIEW

File: service/nodeconfig/v1/resource/node/create.go [hunk 1]
```diff
@@ -59,36 +61,74 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
 		p := []byte(UnschedulablePatch)
 
-		{
-			manifest, err := k8sClient.CoreV1().Nodes().Get(n, apismetav1.GetOptions{})
-			if err != nil {
-				return microerror.Mask(err)
-			}
-			fmt.Printf("unschedulable\n")
-			fmt.Printf("%#v\n", manifest.Spec.Unschedulable)
-			fmt.Printf("unschedulable\n")
```
Comment (xh3b4sd) [inline service/nodeconfig/v1/resource/node/create.go]: Removing the logging here.

File: service/nodeconfig/v1/resource/node/create.go [hunk 2]
```diff
@@ -59,36 +61,74 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
 			return microerror.Mask(err)
 		}
 
-		{
-			manifest, err := k8sClient.CoreV1().Nodes().Get(n, apismetav1.GetOptions{})
-			if err != nil {
-				return microerror.Mask(err)
+		customPods = filterPods(podList.Items, func(p v1.Pod) bool {
+			if p.GetNamespace() == "kube-system" {
+				return false
```
Comment (xh3b4sd) [inline service/nodeconfig/v1/resource/node/create.go]: Filtering for custom pods.

File: service/nodeconfig/v1/resource/node/create.go [hunk 3]
```diff
@@ -59,36 +61,74 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
 			}
-			fmt.Printf("unschedulable\n")
-			fmt.Printf("%#v\n", manifest.Spec.Unschedulable)
-			fmt.Printf("unschedulable\n")
+
+			return true
+		})
+		systemPods = filterPods(podList.Items, func(p v1.Pod) bool {
+			if p.GetNamespace() == "kube-system" {
+				return true
```
Comment (xh3b4sd) [inline service/nodeconfig/v1/resource/node/create.go]: Filtering for system pods.

File: service/nodeconfig/v1/resource/node/create.go [hunk 4]
```diff
@@ -59,36 +61,74 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
...
+		fmt.Printf("\n")
+		fmt.Printf("customPods\n")
+
+		fmt.Printf("\n")
+		fmt.Printf("systemPods\n")
+		for _, p := range systemPods {
+			fmt.Printf("%#v\n", p)
+		}
+		fmt.Printf("\n")
+		fmt.Printf("systemPods\n")
```
Comment (xh3b4sd) [inline service/nodeconfig/v1/resource/node/create.go]: Some debugging here.

File: service/nodeconfig/v1/resource/node/create.go (hunk 2 above)
Comment (kopiczko) [reply service/nodeconfig/v1/resource/node/create.go]: You could do that in single loop w/o filter function and I would find that cleaner.

Comment (xh3b4sd) [reply service/nodeconfig/v1/resource/node/create.go]: Ok cool. Done.

**Comment by xh3b4sd** [comment]
Fishing for LGTM here.

**Pull request merged**